File: frontend/src/types/wardrobe.ts
export interface WardrobeItem {
  id: string
  category: string
  color: string
  label: string | null
  created_at: string
  image_url: string
}

export interface PaginatedWardrobe {
  items: WardrobeItem[]
  total: number
  page: number
  page_size: number
}

File: backend/app/schemas/wardrobe.py
import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class WardrobeItemRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    category: str
    color: str
    label: str | None
    created_at: datetime
    image_url: str


class PaginatedWardrobe(BaseModel):
    items: list[WardrobeItemRead]
    total: int
    page: int
    page_size: int
